x/concentrated-liquidity/position.go

```diff
@@ -547,44 +552,69 @@ func (k Keeper) validatePositionsAndGetTotalLiquidity(ctx sdk.Context, owner sdk
 	return basePosition.PoolId, basePosition.LowerTick, basePosition.UpperTick, totalLiquidity, nil
 }
 
-// GetPositionIdToLock returns the positionId to lock mapping in state.
-func (k Keeper) GetPositionIdToLock(ctx sdk.Context, positionId uint64) (uint64, error) {
+// GetLockIdFromPositionId returns the lock id associated with the given position id.
+func (k Keeper) GetLockIdFromPositionId(ctx sdk.Context, positionId uint64) (uint64, error) {
 	store := ctx.KVStore(k.storeKey)
 
-	// Get the position ID to key mapping.
-	key := types.KeyPositionIdForLock(positionId)
-	value := store.Get(key)
+	// Get the position ID to lock ID mapping.
+	positionIdLockKey := types.KeyPositionIdForLock(positionId)
+	value := store.Get(positionIdLockKey)
 	if value == nil {
 		return 0, types.PositionIdToLockNotFoundError{PositionId: positionId}
 	}
 
 	return sdk.BigEndianToUint64(value), nil
 }
 
-// SetPositionIdToLock sets the positionId to lock mapping in state.
+// GetPositionIdToLockId returns the position id associated with the given lock id.
+func (k Keeper) GetPositionIdToLockId(ctx sdk.Context, underlyingLockId uint64) (uint64, error) {
+	store := ctx.KVStore(k.storeKey)
+
+	// Get the lock ID to position ID mapping.
+	positionIdLockKey := types.KeyLockIdForPositionId(underlyingLockId)
+	value := store.Get(positionIdLockKey)
+	if value == nil {
+		return 0, types.LockIdToPositionIdNotFoundError{LockId: underlyingLockId}
+	}
+
+	return sdk.BigEndianToUint64(value), nil
+}
+
+// SetPositionIdToLock sets both the positionId to lock mapping and the lock to positionId mapping in state.
 func (k Keeper) SetPositionIdToLock(ctx sdk.Context, positionId, underlyingLockId uint64) {
 	store := ctx.KVStore(k.storeKey)
 
-	// Get the position ID to key mapping.
-	key := types.KeyPositionIdForLock(positionId)
-	store.Set(key, sdk.Uint64ToBigEndian(underlyingLockId))
+	// Get the position ID to key mapping and set.
+	positionIdLockKey := types.KeyPositionIdForLock(positionId)
+	store.Set(positionIdLockKey, sdk.Uint64ToBigEndian(underlyingLockId))
+
+	// Get the lock ID to key mapping and set.
+	lockIdKey := types.KeyLockIdForPositionId(underlyingLockId)
+	store.Set(lockIdKey, sdk.Uint64ToBigEndian(positionId))
 }
 
-// RemovePositionIdToLock removes the positionId to lock mapping from state.
+// RemovePositionIdToLock removes both the positionId to lock mapping and the lock to positionId mapping in state.
 func (k Keeper) RemovePositionIdToLock(ctx sdk.Context, positionId uint64) {
 	store := ctx.KVStore(k.storeKey)
 
-	// Get the position ID to key mapping.
-	key := types.KeyPositionIdForLock(positionId)
+	// Get the position ID to lock mapping.
+	positionIdLockKey := types.KeyPositionIdForLock(positionId)
+	underlyingLockId := sdk.BigEndianToUint64(store.Get(positionIdLockKey))
+
+	// Delete the mapping from state.
+	store.Delete(positionIdLockKey)
+
+	// Get the lock ID to key mapping.
+	lockIdKey := types.KeyLockIdForPositionId(underlyingLockId)
```

Comment:
nit: consider grouping these two keys and working with them via single method. I noticed that these two keys would in normal cases, work as pairs. That is, if one is set the the other is set, if one is deleted, the other is deleted. Creating helper methods to deal with these two pair keys might help us with ensuring that we use them together

Reply:
Sorry I am not quite sure I understand. Right now, the helper method of `Remove` removes both lock to position and position to lock, and `Set` sets both lock to position and position to lock. Therefore, utilizing helper methods alone, you cannot remove one pair with out the other, and same for set. Is there a different way I should be grouping them?

Reply:
Apologies if my questions were unclear, I was referring to making a helper method, maybe something called `PositionIdToLockKeys` where it would return an array of key for `KeyPositionIdForLock` and `KeyLockIdForPositionId`. The reason why this might be a useful way is because as of right now, when we write code, we "must" have context or remember manually that these keys are grouped together thus have to be called together. Just a suggestion though, feel free to resolve this comment if you have different thoughts

Reply:
Merged the set and delete methods (kept the getter separated, since there are times you only have one and want the other) https://github.com/osmosis-labs/osmosis/pull/4885/commits/89be60d0bfc3e8c250c1a607e10e83a329a37d4c